src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -20,42 +20,56 @@
 
 namespace OpenTelemetry.Instrumentation.Process;
 
-internal class ProcessMetrics
+internal sealed class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
-    internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
-    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
+    internal readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
 
-    static ProcessMetrics()
+    private readonly Diagnostics.Process currentProcess = Diagnostics.Process.GetCurrentProcess();
+    private double? memoryUsage;
+    private double? virtualMemoryUsage;
+
+    public ProcessMetrics(ProcessInstrumentationOptions options)
     {
         // TODO: change to ObservableUpDownCounter
-        MeterInstance.CreateObservableGauge(
+        this.MeterInstance.CreateObservableGauge(
             "process.memory.usage",
             () =>
             {
-                CurrentProcess.Refresh();
-                return CurrentProcess.WorkingSet64;
+                if (!this.memoryUsage.HasValue)
+                {
+                    this.Snapshot();
+                }
+
+                var value = this.memoryUsage.Value;
+                this.memoryUsage = null;
+                return value;
             },
             unit: "By",
-            description: "The amount of physical memory in use.");
+            description: "The amount of physical memory allocated for this process.");
 
         // TODO: change to ObservableUpDownCounter
-        MeterInstance.CreateObservableGauge(
+        this.MeterInstance.CreateObservableGauge(
             "process.memory.virtual",
             () =>
             {
-                CurrentProcess.Refresh();
-                return CurrentProcess.VirtualMemorySize64;
+                if (!this.virtualMemoryUsage.HasValue)
+                {
+                    this.Snapshot();
+                }
+
+                var value = this.virtualMemoryUsage.Value;
+                this.virtualMemoryUsage = null;
+                return value;
             },
             unit: "By",
-            description: "The amount of committed virtual memory.");
+            description: "The amount of virtual memory allocated for this process that cannot be shared with other processes.");
```

Comment:
I was expecting "process.memory.virtual" to be Process.VirtualMemorySize64. I saw you linked to https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/673 which had a very nice explanation for why you made the change. I think you were attempting to be very thorough and precise in making this choice so I want to offer something equally detailed to give an alternate perspective on the problem.

When we get into details, I think 'committed memory' is unfortunately an ambiguous term and the best way to understand what the OTel spec writers intended will be for us to ask them and clarify the spec. In Mark Russinovich's blog post he repeatedly refers to memory that "counts against the commit limit", but he never actually says "Memory is 'committed memory' if and only if it counts against the commit limit". I think that would be an intuitive definition in context of his article and probably it is the definition he intended the reader to apply despite never stating it pedantically. For simplicity we can just call that his intended definition. His definition is probably technically precise for how some portion of the Windows Kernel uses the term, but I suspect it is not how most people understand and use it.

IMO the common definition of committed memory that most developers would use is that it is all the memory addresses a process can read from without triggering an access violation. This includes (at least) three categories of memory:
 - memory backed solely by RAM
 - memory backed by a swapfile/pagefile
 - memory backed by some other mapped file on disk

The first two agree with Mark's definition above so the difference is entirely whether the third bullet is included. To justify that it is common for most people to include file backed memory as part of 'committed memory' [this article](https://dzone.com/articles/windows-process-memory-usage-demystified) has a screenshot of the tool VMMap at the top. I consider VMMap to be the de-facto standard for investigating and categorizing virtual memory usage (incidentally its also a tool MarkR made). Looking at the table in that image is a column called 'Committed' and the first two rows are 'Image' and 'Mapped File'. If committed virtual memory did not include memory backed by files on disk then we'd expect those columns to be 0, but they aren't.

As a second example, [this link](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/-address) is a help page for the !address command in windbg which shows information about the virtual address space. There is an example output part way down the page showing:
```
0:000> !address 75831234
Usage:                  Image
Base Address:           75831000
End Address:            758f6000
Region Size:            000c5000
Type:                   01000000MEM_IMAGE
State:                  00001000MEM_COMMIT
Protect:                00000020PAGE_EXECUTE_READ
More info:              lmv m kernel32
More info:              !lmi kernel32
More info:              ln 0x75831234
```

The debugger says that this address is part of the kernel32 image and it is on a memory page that is marked both as being MEM_IMAGE and MEM_COMMIT. Mark's blog post says specifically that "The virtual memory in Testlimit's address space where its executable and system DLL images are mapped therefore don't count toward the commit limit" and yet windbg describes this exact type of memory as being MEM_COMMIT.

I think the way VMMap and Windbg refer to committed memory is the way most developers would use the term. That definition corresponds to using Process.VirtualMemorySize64 as the metric value. Of course we can always ask the OTel spec writers if that is the definition they intended. Hope that helps!